### PR TITLE
Add structure for Messenger Send API

### PR DESCRIPTION
**Pull request by purelogiq**
The [Messenger Send API](https://developers.facebook.com/docs/messenger-platform/send-api-reference) is somewhat complex or at the very least hard to memorize. This PR is aimed at adding some basic structure to the app to help us build the rest of it moving forward.

In the `services` directory, you'll find a `senders` folder which are helper classes that correspond to one of Messenger Send API features. They all inherit from the `AbstractSender` which takes care of some general stuff.

I went ahead and implemented the `MessageSender` (for plain messages), and the `MediaSender` (for images, video, audio, files).

Please let me know what you think.

### CODE REVIEW

File: app/services/senders/media_sender.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  def set_audio(audio)
+    set_media('audio', audio)
+  end
+
+  def set_video(video)
+    set_media('video', video)
+  end
+
+  def set_file(file)
```
[the remark this answers is not in the document]
Comment (purelogiq) [reply app/services/senders/media_sender.rb:22]: Resolved by disabling this check :P

File: app/services/senders/media_sender.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  def set_image(image)
+    set_media('image', image)
+  end
+
+  def set_audio(audio)
+    set_media('audio', audio)
+  end
+
+  def set_video(video)
```
[the remark this answers is not in the document]
Comment (purelogiq) [reply app/services/senders/media_sender.rb:18]: Resolved by disabling this check :P

File: app/services/senders/media_sender.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+class MediaSender < AbstractSender
+  SITE_DOMAIN = 'https://letsthryve.com'
+
+  def set_image(image)
+    set_media('image', image)
+  end
+
+  def set_audio(audio)
```
[the remark this answers is not in the document]
Comment (purelogiq) [reply app/services/senders/media_sender.rb:14]: Resolved by disabling this check :P

File: app/services/senders/media_sender.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+class MediaSender < AbstractSender
+  SITE_DOMAIN = 'https://letsthryve.com'
+
+  def set_image(image)
```
[the remark this answers is not in the document]
Comment (purelogiq) [reply app/services/senders/media_sender.rb:10]: Resolved by disabling this check :P

File: app/services/senders/card_sender.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  #         .add_share_button
+  # sender.add_element(...)
+  # sender.deliver!
+
+  def initialize(recipient)
+    super(recipient)
+    @elements = []
+  end
+
+  def add_element(title: nil, subtitle: nil, item_url: nil, image_url: nil)
```
[the remark this answers is not in the document]
Comment (purelogiq) [reply app/services/senders/card_sender.rb]: Hound is sortof right on the first one, but, this just the most straightforward way to do this (other than creating a button class).

Up'ed the line length for methods to 50.

:grimacing:

**Review by jeanineh — APPROVED** [review]
lgtm

**Review by ashecheerios — APPROVED** [review]

**Pull request merged**